packages/api-reference/src/hooks/useReferenceLayout.ts

```diff
@@ -0,0 +1,18 @@
+import { type InjectionKey, inject } from 'vue'
+
+/**
+ * The layout of the API reference
+ *
+ * @default 'modern'
+ */
+export type ReferenceLayout = 'modern' | 'classic'
+export const REFERENCE_LAYOUT_SYMBOL = Symbol() as InjectionKey<ReferenceLayout>
+
+/**
+ * Get the current API reference layout
+ *
+ * @see {@link ApiReferenceLayout}
+ */
+export const useReferenceLayout = () => ({
```

Comment:
I believe the new style we have for these stateful hooks is call with params for provide, then call with no params for inject. Right @hanspagel ?

But also I think we should do this sparingly as its annoying to test. Maybe we combine everything into the workspace store instead? Then we only need to worry about one store for testing instead of multiple things.

cc: @hanspagel @DemonHa 

tagging @DemonHa so we can come up with a strategy for adding these things into the workspace store. Probably just an extension on the document eh?

Reply:
Yeah this is a legacy way to do this. x-scalar-layout can be added to workspace JSON def and consumed. 

Props can be drilled down as needed. This scattered approach for state management is just not robust.

Reply:
> Right @hanspagel ?

Right @amritk 👍